src/picongpu/include/traits/UsesRNG.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2016 Marco Garten, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/mpl/bool.hpp>
+
+namespace picongpu
+{
+namespace traits
+{
+
+/** Checks if an object requires the RNG
+ *
+ * @tparam T_Object any object (class or typename)
+ *
+ * This struct must define
+ * ::type (boost::mpl::bool_<>)
+ */
+template<typename T_Object>
+struct UsesRNG
```

Comment:
what @Flamefire used are static const bool integrals and in the enum types which are by definition "nullary" constants (but not [nullary meta functions](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/metafunction.html)).

In the strict mpl context, [`boost integral constants`](http://www.boost.org/doc/libs/1_61_0/libs/type_traits/doc/html/boost_typetraits/reference/integral_constant.html) as in [`is_unsigned`](http://www.boost.org/doc/libs/1_61_0/libs/type_traits/doc/html/boost_typetraits/reference/is_unsigned.html) are not identical to [`boost::mpl integral types`](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/integral-constant.html) (e.g. [`mpl::bool_`](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/bool.html)).

For the predicate `Pred`  of [`mpl _if Transformations`](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/transformation-algorithms.html) a *Unary [Lambda Expression](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/lambda-expression.html)* is required, which is either a metafunction (not an integral constant!) or a lambda.

Nevertheless, as soon as a strict meta function is expected, e.g., in the context of [`mpl::apply`](http://www.boost.org/doc/libs/1_61_0/libs/mpl/doc/refmanual/apply.html) the missing requirement of a nullary, `::type`-able meta-function will cause trouble. The thing that saves his example here again is, that the integral constant is wrapped in a lambda which is `mpl::apply`-able.